src/integrations/front-end/print-qrcode-render.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Yoast\WP\SEO\Integrations\Front_End;
+
+use chillerlan\QRCode\QRCode;
+use chillerlan\QRCode\QROptions;
+use Yoast\WP\SEO\Conditionals\Front_End_Conditional;
+use Yoast\WP\SEO\Conditionals\Print_QR_Code_Enabled_Conditional;
+use Yoast\WP\SEO\Integrations\Integration_Interface;
+
+/**
+ * Class that renders a QR code for URLs.
+ */
+class Print_QRCode_Render implements Integration_Interface {
+
+	/**
+	 * Register the hooks.
+	 *
+	 * @return void
+	 */
+	public function register_hooks() {
+		\add_action( 'send_headers', [ $this, 'generate_qr_code' ], 0 );
+	}
+
+	/**
+	 * Returns the conditionals based in which this loadable should be active.
+	 *
+	 * @return array
+	 */
+	public static function get_conditionals() {
+		return [ Front_End_Conditional::class, Print_QR_Code_Enabled_Conditional::class ];
+	}
+
+	/**
+	 * Generates a QR code and outputs it as SVG.
+	 *
+	 * @return void
+	 */
+	public function generate_qr_code() {
+		$url = \filter_input( INPUT_GET, 'yoast_qr_code', FILTER_SANITIZE_URL );
```

Comment:
We should probably also validate that it's an internal URL and not pointing to another domain.

I don't see any real security implications here but it's a simple check and better to be safe than sorry.

Reply:
That's true, let's leave it as is then.